### PR TITLE
Use ingest endpoint instead of api endpoint to send events

### PR DESCRIPTION
**Pull request by udaysagar2177**
Update json events format and protobuf messages to support V2 events
pep8 compliant code
Fix mdl warnings in README.md
Add circle.yml
Update README.md
Update .gitignore

### CODE REVIEW

**Comment by udaysagar2177** [comment]
I will squash the commits before merging!

File: README.md [hunk 1]
```diff
@@ -204,7 +205,8 @@ atexit.register(sfx.stop)
 
 ```
 ERROR:root:Posting to SignalFx failed.
-SSLError: hostname 'api.signalfx.com' doesn't match either of '*.signalfuse.com', 'signalfuse.com'
+SSLError: hostname 'ingest.signalfx.com' doesn't match either of
```
Comment (ramjothikumar) [inline README.md]: @udaysagar2177 Is this still valid. Given we moved from api to ingest, this limitation might not be there anymore. Can you check and update this please.

**Comment by udaysagar2177** [comment]
Ingest endpoint is using https, so I think it is still valid.

**Comment by ramjothikumar** [comment]
@udaysagar2177 So this was happening only for events earlier as the issue was with the api endpoint. The api call to ingest (https) did not have any issues. Can you try this and check it out on a virtualenv env.

**Pull request merged**